sunpy/instr/aia/response_utils.py

```diff
@@ -0,0 +1,228 @@
+                    print('Setting {} to 1'.format(prop[1]))
+                    row[prop[1]] = 1
+                else:
+                    row[prop[1]] = channel_data[prop[0].upper()]
+            rows.append(row)
+
+    # assign units
+    table = QTable(rows=rows,
+                   names=tuple(['channel']+[p[1] for p in properties]))
+    for name in table.colnames:
+        try:
+            table[name].unit = units[name]
+        except TypeError:
+            # weird astropy table exception, units still set in all cases
+            # exception seems to be only thrown when reading in the UV channels
+            pass
+
+    return table
+
+
+@u.quantity_input(temperature=u.K, density=u.cm**(-3), continuum_wavelength=u.angstrom)
+def make_emiss_table(emiss_table_file, ion_list, temperature=np.logspace(5, 8, 50)*u.K,
+                     density=1e15/np.logspace(5, 8, 50)*u.cm**(-3),
+                     continuum_wavelength=np.arange(10., 500., 1.)*u.angstrom, **kwargs):
+    """
+    Build line and continuum emissivity table.
+
+    Build HDF5 file of line and continuum contributions for each ion. For each ion, the
+    contribution function as well as the three components of the continuum, two-photon,
+    free-free, and free-bound, are calculated. Note that the contribution function is calculated
+    here as,
+
+    .. math::
+       G(\lambda,n,T) = \\frac{1}{4\pi}\mathrm{Ab}(Z)\\frac{N(Z^{+z})}{N(Z)}\epsilon(\lambda,n,T)\\frac{1}{n_e},
+
+    in units of photon :math:`\mathrm{cm}^{3}\,\mathrm{s}^{-1}\,\mathrm{str}^{-1}`. See the
+    ChiantiPy documentation for more information regarding the continuum calculation.
+
+    Parameters
+    ----------
+    emiss_table_file : `str`
+    ion_list : `list`
+        List of ion names, e.g. 'fe_15' for Fe XV
+    temperature : array-like, optional
+    density : array-like, optional
+        Default is to assume a constant pressure of :math:`10^{15}\,\mathrm{K}\,\mathrm{cm}^{-3}`
+    continuum_wavelength : array-like, optional
+
+    See also
+    --------
+    EmissTableInterface : Easily access data in emissivity table files.
+    """
+
+    ch_data.Defaults['flux'] = 'photon'
+    abundance_file = kwargs.get('abundance_file', 'sun_coronal_1992_feldman')
+    ch_data.Defaults['ioneqfile'] = kwargs.get('ionization_equilibrium_file', ch_data.Defaults['ioneqfile'])
+
+    with h5py.File(emiss_table_file, 'x') as hf:
+        # save parameters
+        dset_temperature = hf.create_dataset('temperature', data=temperature.value)
+        dset_temperature.attrs['unit'] = temperature.unit.to_string()
+        dset_density = hf.create_dataset('density', data=density.value)
+        dset_density.attrs['unit'] = density.unit.to_string()
+        dset_wavelength = hf.create_dataset('continuum_wavelength', data=continuum_wavelength.value)
+        dset_wavelength.attrs['unit'] = continuum_wavelength.unit.to_string()
+
+        for ion in ion_list:
```

Comment:
I like that. I considered doing some logging/print statements to show progress, but a progress bar is neater.